components/script_plugins/webidl_must_inherit.rs

```diff
@@ -0,0 +1,235 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use rustc::hir::{self, HirId};
+use rustc::lint::{LateContext, LateLintPass, LintArray, LintContext, LintPass};
+use rustc::ty;
+use std::boxed;
+use std::env;
+use std::error::Error;
+use std::fmt;
+use std::fs;
+use std::io;
+use std::path;
+use syntax::ast;
+use webidl::ast::*;
+use webidl::visitor::*;
+use webidl::*;
+
+declare_lint!(
+    WEBIDL_INHERIT_CORRECT,
+    Deny,
+    "Warn and report usage of incorrect webidl inheritance"
+);
+
+pub struct WebIdlPass;
+
+#[derive(Clone, Debug)]
+pub enum WebIdlError {
+    ParentMismatch {
+        name: String,
+        rust_parent: String,
+        webidl_parent: String,
+    },
+}
+
+impl fmt::Display for WebIdlError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
```

Comment:
It is sort of weird seeing a match here being used solely for destructuring. You can actually instead do the following:
```rust
let WebIdlError { name, rust_parent, webidl_parent } = self;
```
or simply just use `self.name`, `self.rust_parent` and `self.webidl_parent` without destructuring.